optimum/neuron/modeling_base.py

```diff
@@ -12,600 +12,29 @@
 
-from ..exporters.neuron import main_export
-from ..exporters.neuron.model_configs import *  # noqa: F403
-from ..exporters.tasks import TasksManager
 from ..modeling_base import OptimizedModel
-from .utils import (
-    NEURON_FILE_NAME,
-    check_if_weights_replacable,
-    is_neuron_available,
-    replace_weights,
-    store_compilation_config,
-)
-from .utils.hub_neuronx_cache import ModelCacheEntry, build_cache_config, create_hub_compile_cache_proxy
-from .utils.import_utils import is_neuronx_available
-from .utils.misc import maybe_load_preprocessors
-from .utils.version_utils import check_compiler_compatibility, get_neuroncc_version, get_neuronxcc_version
 
 
 if TYPE_CHECKING:
-    from transformers import PretrainedConfig
+    from transformers import PretrainedConfig, PreTrainedModel
 
-    from ..exporters.neuron import NeuronDefaultConfig
 
-if is_neuron_available():
-    NEURON_COMPILER_TYPE = "neuron-cc"
-    NEURON_COMPILER_VERSION = get_neuroncc_version()
+class NeuronModel(OptimizedModel):
 
-if is_neuronx_available():
-    NEURON_COMPILER_TYPE = "neuronx-cc"
-    NEURON_COMPILER_VERSION = get_neuronxcc_version()
-
-logger = logging.getLogger(__name__)
-
-
-class NeuronBaseModel(OptimizedModel):
-    """
-    Base class running compiled and optimized models on Neuron devices.
-
-    It implements generic methods for interacting with the Hugging Face Hub as well as compiling vanilla
-    transformers models to neuron-optimized TorchScript module and export it using `optimum.exporters.neuron` toolchain.
-
-    Class attributes:
-        - model_type (`str`, *optional*, defaults to `"neuron_model"`) -- The name of the model type to use when
-        registering the NeuronBaseModel classes.
-        - auto_model_class (`Type`, *optional*, defaults to `AutoModel`) -- The `AutoModel` class to be represented by the
-        current NeuronBaseModel class.
-
-    Common attributes:
-        - model (`torch.jit._script.ScriptModule`) -- The loaded `ScriptModule` compiled for neuron devices.
-        - config ([`~transformers.PretrainedConfig`]) -- The configuration of the model.
-        - model_save_dir (`Path`) -- The directory where a neuron compiled model is saved.
-        By default, if the loaded model is local, the directory where the original model will be used. Otherwise, the
-        cache directory will be used.
-    """
-
-    model_type = "neuron_model"
-    auto_model_class = AutoModel
-    library_name = "transformers"
-
-    def __init__(
-        self,
-        model: torch.jit._script.ScriptModule,
-        config: "PretrainedConfig",
-        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
-        model_file_name: Optional[str] = None,
-        preprocessors: Optional[List] = None,
-        neuron_config: Optional["NeuronDefaultConfig"] = None,
-        **kwargs,
-    ):
+    def __init__(self, model: "PreTrainedModel", config: "PretrainedConfig"):
```

Comment:
For traced models, we are passing `torch.jit._script.ScriptModule` as model here.

Reply:
I realized that when the CI failed ... 🫣